tests/providers/amazon/aws/operators/test_emr_containers.py

```diff
@@ -144,6 +145,14 @@ def test_execute_with_polling_timeout(self, mock_check_query_status):
             assert "Final state of EMR Containers job is SUBMITTED" in str(ctx.value)
             assert "Max tries of poll status exceeded" in str(ctx.value)
 
+    @mock.patch.object(EmrContainerHook, "submit_job")
+    def test_operator_defer(self, mock_submit_job):
+        self.emr_container.deferrable = True
+        self.emr_container.wait_for_completion = False
+        with pytest.raises(TaskDeferred) as exc:
+            self.emr_container.execute(context=None)
+        assert isinstance(exc.value.trigger, EmrContainerTrigger), "Trigger is not a EmrContainerTrigger"
```

Comment:
Perhaps we can enhance the message by including the trigger in the print.

```suggestion
        assert isinstance(exc.value.trigger, EmrContainerTrigger), f"{exc.value.trigger} is not a EmrContainerTrigger"
```